### PR TITLE
Added support for case management and event listing

### PR DESCRIPTION
**Pull request by yaakovi**
fixes: https://github.com/demisto/server/issues/8706
related: demisto/server#8751

### CODE REVIEW

File: Integrations/integration-ESM-v10.yml [hunk 1]
```diff
@@ -302,17 +506,182 @@ script:
...
+            fixed_users.append({
+                'ID' : user['id']['value'],
+                'Name' : user['username'],
+                'Email' : user['email'],
+                'SMS' : user['sms'],
+                'Is Master' : user['master'],
+                'Is Admin' : user['admin'],
+            })
+
+        context = {'User(val.ID && val.ID == obj.ID)' : fixed_users}
```
Comment (asafshen) [inline Integrations/integration-ESM-v10.yml]: I think it should be `EsmUser` instead `User`
Plus - don't forget to update command output

**Comment by meirwah** [comment]
@yaakovi seems the ESM test fails:
```
McAfeeESMTest failed with error/s
Playbook McAfeeESMTest has failed:
	- Command: !esm-add-case-status name="Research" show_in_case_pane="True"
	- Body: Failed login
url: https://37.19.123.148:35004/rs/esm/login
 response status: 401
response: {
  "code" : "2",
  "message" : "ERROR_MaxSessionsHit (2)",
  "errors" : [ ]
}

	- Body: Playbook **McAfeeESMTest** execution error
```

**Comment by meirwah** [comment]
https://circleci.com/gh/demisto/content/3524?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link

File: Integrations/integration-ESM-v10.yml [hunk 2]
```diff
@@ -264,19 +296,236 @@ script:
...
+            status_id = self.case_status_name_to_id(name)
+            status_id = {'value' : status_id}
+
+            cmd = 'caseDeleteCaseStatus'
+            query = json.dumps({'statusId' : status_id})
+            self.cmdquery(cmd, query, no_answer=True)
+
+            return 'Deleted case status with ID: %d' % (status_id['value'], )
+
+        def get_case_statuses(self, should_login = False): # TODO: consider using cache for the mapping id<->name
```
Comment (asafshen) [inline Integrations/integration-ESM-v10.yml]: please remove TODO

File: Integrations/integration-ESM-v10.yml [hunk 3]
```diff
@@ -302,17 +551,229 @@ script:
...
+        last_run = demisto.getLastRun()
+        # for backward compatibility uses
+        if last_run.has_key('value') and not last_run.has_key('alarms'):
+            last_run['alarms'] = last_run['value']
+
+        start_alarms = last_run.get('alarms', (datetime.datetime.now() - datetime.timedelta(minutes=1)).isoformat())
+        start_cases = last_run.get('cases', int(demisto.params()['startingCaseID']))
+
+        incidents = []
+        mode = demisto.params()['fetchTypes'].lower()
```
Comment (asafshen) [inline Integrations/integration-ESM-v10.yml]: If user upgrade existing instance from old ESM to this ESM - I think `fetchTypes` will not get the default value of `alarms`

I think we should have fallback to `alarms` also here

**Pull request merged**